front/src/common/Tabs.tsx

```diff
@@ -30,13 +39,22 @@ const Tab = ({ label, content }: TabComponentProps) => (
 
 const Tabs = ({ tabs, pills = false, fullWidth = false, fullHeight = false }: TabsProps) => {
   const [activeTabIndex, setActiveTabIndex] = useState(0);
+  const { t } = useTranslation();
+  const currentCategory = useSelector(getCategory);
+  const rollingStockId = useSelector(getOperationalStudiesRollingStockID);
+  const { rollingStock } = useStoreDataForRollingStockSelector({
+    rollingStockId,
+  });
+  const showWarning =
+    rollingStock && currentCategory && !rollingStock?.other_categories.includes(currentCategory);
```

Comment:
Be carreful with the `?`, you should only use them if the object you're manipulating can be `null` or `undefined`, which is not the case here ;)
```suggestion
    rollingStock && currentCategory && !rollingStock.other_categories.includes(currentCategory);
```

Reply:
Done !